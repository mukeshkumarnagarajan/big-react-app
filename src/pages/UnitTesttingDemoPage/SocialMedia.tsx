import { ChangeEvent, useState } from "react";

interface SocialMediaProps {
  name: string;
  followersCount: string;
}

const SocialMedia: React.FC<SocialMediaProps> = ({ name, followersCount }) => {
  const [countryName, setCountryName] = useState("India");

  return (
    <>
      <h2>Social Media Info</h2>
      <h3>
        Demo of Testing Props, States, Events, JSX, Styles, Snapshot testing
      </h3>

      <p>Top Social Media Platform of Genesys</p>
      <p data-testid="followersDetails" style={{backgroundColor: '#ffff00'}}>
        {name} with {followersCount} followers
      </p>

      <input
        type="text"
        placeholder="Enter Country Name"
        value={countryName}
        onChange={(event: ChangeEvent<HTMLInputElement>) => {
          setCountryName(event.target.value);
        }}
      />
      <p data-testid="webSiteInfo">Visit Genesys {countryName} Website</p>
    </>
  );
};

export default SocialMedia;
