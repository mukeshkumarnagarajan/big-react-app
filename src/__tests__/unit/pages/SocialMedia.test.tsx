import { fireEvent, render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import SocialMedia from "../../../pages/UnitTesttingDemoPage/SocialMedia";

it("has Social Media in heading", () => {
  render(<SocialMedia name="Facebook" followersCount="2.3M+" />);
  const headingElement = screen.getByText(/Social Media Info/i);
  expect(headingElement).toBeDefined();
});

it("receives the props name, followersCount and displays in paragraph", () => {
  render(<SocialMedia name="Facebook" followersCount="2.3M+" />);
  const followersDetailElement = screen.getByTestId("followersDetails");
  expect(followersDetailElement.textContent).toBe(
    "Facebook with 2.3M+ followers"
  );
});

it('should have input field with placeholder "Enter Country Name"', () => {
  render(<SocialMedia name="Facebook" followersCount="2.3M+" />);
  const countryInput = screen.getAllByPlaceholderText("Enter Country Name");
  expect(countryInput).toBeTruthy();
});

it("should have input field with default value India", () => {
  render(<SocialMedia name="Facebook" followersCount="2.3M+" />);
  const countryInput: HTMLInputElement = screen.getByRole("textbox");
  expect(countryInput.value).toBe("India");
});

it("should update country name on change event", () => {
  render(<SocialMedia name="Facebook" followersCount="2.3M+" />);
  const countryInput: HTMLInputElement = screen.getByRole("textbox");

  fireEvent.change(countryInput, {
    target: {
      value: "Singapore",
    },
  });

  expect(countryInput.value).toBe("Singapore");

  const webSiteInfoElement = screen.getByTestId("webSiteInfo");
  expect(webSiteInfoElement.textContent).toBe(
    "Visit Genesys Singapore Website"
  );
});

it("displays social media followers count with background color green", () => {
  render(<SocialMedia name="Facebook" followersCount="2.3M+" />);
  const followersDetailElement = screen.getByTestId("followersDetails");
  expect(followersDetailElement).toHaveStyle("background-color:#ffff00");
});

it("has right SocialMedia component snapshot", () => {
  const snapshotJson = renderer.create(
    <SocialMedia name="Facebook" followersCount="2.3M+" />
  );
  expect(snapshotJson).toMatchSnapshot();
});
