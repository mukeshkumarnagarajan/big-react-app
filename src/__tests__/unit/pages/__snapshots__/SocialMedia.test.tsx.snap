// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`has right SocialMedia component snapshot 1`] = `
[
  <h2>
    Social Media Info
  </h2>,
  <h3>
    Demo of Testing Props, States, Events, JSX, Styles, Snapshot testing
  </h3>,
  <p>
    Top Social Media Platform of Genesys
  </p>,
  <p
    data-testid="followersDetails"
    style={
      {
        "backgroundColor": "#ffff00",
      }
    }
  >
    Facebook
     with 
    2.3M+
     followers
  </p>,
  <input
    onChange={[Function]}
    placeholder="Enter Country Name"
    type="text"
    value="India"
  />,
  <p
    data-testid="webSiteInfo"
  >
    Visit Genesys 
    India
     Website
  </p>,
]
`;
